### PR TITLE
Bug fix: decode bytes object to string for subprocess.check_output ou…

### PR DESCRIPTION
**Pull request by dapeng-mi**
When running test "./tools/run_tests/run_tests.py -l c++ -c dbg" in python3
environment, the following exception is observed.

```
./tools/run_tests/run_tests.py -l c++ -c dbg
PASSED: make [time=114.9sec, retries=0:0]

Omitting EPOLLEXCLUSIVE tests

2020-12-01 10:45:40,806 detected port server running version 21
2020-12-01 10:45:40,866 my port server is version 21
Traceback (most recent call last):
  File "./tools/run_tests/run_tests.py", line 1919, in <module>
    build_only=args.build_only)
  File "./tools/run_tests/run_tests.py", line 1800, in _build_and_run
    one_run = set(spec for language in languages
  File "./tools/run_tests/run_tests.py", line 1801, in <genexpr>
    for spec in language.test_specs()
  File "./tools/run_tests/run_tests.py", line 383, in test_specs
    for line in tests.split('\n'):
TypeError: a bytes-like object is required, not 'str'
```

### CODE REVIEW

**Comment by dapeng-mi** [comment]
any comments?

**Comment by jtattermusch** [comment]
Adding @gnossen  as this is related to python2 -> python3 move for our test scripts.

File: tools/run_tests/run_tests.py [hunk 1]
```diff
@@ -355,7 +355,7 @@ def test_specs(self):
                             tests = subprocess.check_output(
                                 [binary, '--benchmark_list_tests'],
                                 stderr=fnull)
-                        for line in tests.split('\n'):
+                        for line in tests.decode().split('\n'):
```
Comment (jtattermusch) [inline tools/run_tests/run_tests.py:358]: should or should we not specify the encoding here explicitly?

Comment (jtattermusch) [inline tools/run_tests/run_tests.py:358]: we need to make sure this still works under python2.7 as well (AFAIK some parts of the infrastructure still use it) - have you tested that?

**Review by jtattermusch — COMMENTED** [review]
Thanks for the fix! It looks alright, but I had some questions.

I'll leave the final approval to @gnossen

File: tools/run_tests/run_tests.py (hunk 1 above)
Comment (gnossen) [reply tools/run_tests/run_tests.py:358]: [The default is "UTF-8"](https://docs.python.org/3/library/stdtypes.html#bytes.decode). This should be fine.

Comment (gnossen) [reply tools/run_tests/run_tests.py:358]: The test runner is still Python 2, so CI will take care of that.

**Review by gnossen — APPROVED** [review]
Thanks for the contribution!

**Comment by dapeng-mi** [comment]
@gnossen @jtattermusch Thanks for reviewing.

**Comment by dapeng-mi** [comment]
@gnossen @jtattermusch Thanks for reviewing. 

Another exception is also found in python3 environment. I submit a commit 965aa55 to fix this expectation. Please help to review again.

File: CMakeLists.txt [hunk 2]
```diff
@@ -223,6 +223,20 @@ endif()
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${_gRPC_C_CXX_FLAGS}")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${_gRPC_C_CXX_FLAGS}")
 
+#if(NOT CMAKE_BUILD_TYPE)
```
Comment (jtattermusch) [inline CMakeLists.txt:226]: please revert this change.

**Review by jtattermusch — CHANGES_REQUESTED** [review]
While the first two python-related commits make sense to me, but I don't understand the purpose of the third commit (and it also doesn't match this PR's intent).
https://github.com/grpc/grpc/pull/24861/commits/83678e0bcfed4474b0a6886de398a664f62a99c0

Please remove the third unrelated commit and then we can do a quick re-review and merge.
If you think the changes in the third commit are useful, please open a separate PR for them.

File: CMakeLists.txt (hunk 2 above)
Comment (dapeng-mi) [reply CMakeLists.txt:226]: This is just for a internal test. Would revert this.

Comment (dapeng-mi) [reply CMakeLists.txt]: The third commit has been reset. Sorry to waste your time, I didn't realize the commit would be sync to this.

**Review by jtattermusch — APPROVED** [review]
LGTM, thanks!
